frontend/src/metabase/query_builder/actions/querying.ts

```diff
@@ -71,18 +73,24 @@ const loadCompleteUIControls = createThunkAction(
   },
 );
 
-export const runDirtyQuestionQuery = () => async (dispatch, getState) => {
-  const areResultsDirty = getIsResultDirty(getState());
-  const queryResults = getQueryResults(getState());
-  const hasResults = !!queryResults;
+export const runDirtyQuestionQuery =
+  () => async (dispatch: Dispatch, getState: GetState) => {
+    const areResultsDirty = getIsResultDirty(getState());
+    const queryResults = getQueryResults(getState());
+    const hasResults = !!queryResults;
 
-  if (hasResults && !areResultsDirty) {
-    const question = getQuestion(getState());
-    return dispatch(queryCompleted(question, queryResults));
-  }
+    if (hasResults && !areResultsDirty) {
+      const question = getQuestion(getState());
 
-  return dispatch(runQuestionQuery());
-};
+      if (!question) {
+        return;
+      }
+
+      return dispatch(queryCompleted(question, queryResults));
```

Comment:
It'd be great if we could get use an actual type instead of `any` for the 2nd argument of `queryCompleted`.